kotlin/workflow-runtime/src/main/java/com/squareup/workflow/internal/LifetimeTracker.kt

```diff
@@ -89,12 +89,15 @@ internal class LifetimeTracker<FactoryT, KeyT : Any, DisposableT>(
     key: KeyT,
     factory: FactoryT
   ): DisposableT {
-    return if (key !in factoriesByKey) {
-      factoriesByKey[key] = factory
+    val existingFactory = factoriesByKey.put(key, factory)
```

Comment:
Semantic change is just pulling this statement up from the first `if` case below, so it's _always_ updated. This is the actual fix.